server/service/src/invoice/mod.rs

```diff
@@ -183,6 +186,17 @@ pub trait InvoiceServiceTrait: Sync + Send {
     ) -> Result<BatchPrescriptionResult, RepositoryError> {
         batch_prescription(ctx, input)
     }
+
+    fn generate_outbound_return_lines(
+        &self,
+        ctx: &ServiceContext,
+        store_id: &str,
+        stock_line_ids: Vec<String>,
+        item_id: Option<String>,
+        return_id: Option<String>,
```

Comment:
Was seeing a bit of both - is there a newer/preferred pattern to pass an input struct vs many params?

Reply:
Apart from store_id i prefer using 'input', benifits kind of small but here they are:

* less things to change if new input is added (just change the to_domain mapping)
* less chance of mapping error when dealing with the same type if destructure/restructure is used, i.e.


```rust

struct ServiceInput {
    id_for_this: String,
    id_for_that: String
}

struct GraphqlInput {
    id_for_this: String,
    id_for_that: String
}

impl GraphqlInput {
    fn to_domain(self) -> ServiceInput {
        GraphqlInput {
            id_for_this,
            id_for_that
        } = self;

        ServiceInput {
            id_for_this,
            id_for_that
        }
    }
}

```

vs

```rust
fn service(id_for_this: String, id_for_that: String) { /* */ }

// in Graphql, when mapping

service(input.id_for_that, input.id_for_this) // got this and that mixed up
```